### DISCUSSION TITLE
NextJs-13

### DISCUSSION THREAD

**Discussion by KSrahul** [vercel/next.js#46146] [Help]
**nextjs-13 is so confusing till now.

NextJs-13, says by default everything inside of app folder is a server component, and can't do client-side interactivity, but almost every page needs a client-side event like onClick, onScroll,

let's say, I have a blog page and that blog data is coming from a server-side/server-component, and there I need a "Load more" button and onClick of that I have to update or add more data in it,

so just for that button click, I have to make another file and make it 'use client', and include it in the server component? Don't you think it's too much, for almost every page

Almost every server-side page needs a client-side activity.

For just that button click inside of the blog page which is server side, I have to make another file and include it in the server component?

And we had everything in nextJs-12, cookie, URL object, res, req, in a single parameter which is "context" for server-side rendering, "getServerSideProps".

Let's say I want to fetch data based on full URL, context.resolvedUrl, which is possible in nextjs-12, "getServerSideProps"

For nextjs-12

In a single page, I do server-side rendering, and let's say, 
I have a blog page and that blog data is coming from a server-side using "getServerSideProps", and I added there a "Load more" button therem and onClick of that I do update or add more data in it,

Pls do something like "getServerSideProps"

According to me, you should take care the especially the below points for nextJs-13, 

1. Instantly page navigate, who use "getServerSideProps", or add a loader till data is loaded, 

2. Add a global server-side data fetching, which means only in a single file who does like "getServerSideProps/server-side" rendering

So I love nextjs-12 till now, but not nextJs-13 till now**

**Comment by arthureberledev** [comment]
You can do everything from nextjs-12 in nextjs-13, you know that you dont have to use the app dir in 13 right? So it is more 13 app dir vs page dir.

For the example with the button, since there is usually a reusable button component in almost every app ive seen, you would have it in a seperate file anyways. Then you just mark this one with "use client" and voila

**Reply by KSrahul** [reply]
So let's say, I use page dir, for pages, and is that page will be server-side rendering? if yes, what should I use there for server-side rendering, using "getServerSideProps" ? or is anything new for server-side rendering inside of page dir in nextjs-13 ?

**Reply by arthureberledev** [reply]
NextJs 13 page dir is as I believe 99% the same as in NextJs 12

**Reply by icyJoseph** [reply]
Pretty much. The `appDir` is a totally separated tech stack, which is in beta state at the moment, so learning it, is not necessary, but playing with it might be a good idea. If it doesn't quite work as the docs, or others are saying it should, let it be and check back later.

The `pages` dir is the good ol' reliable solution it has always been. Nothing has changed in terms of data fetching GSSP, GSP, etc,... though now you can use edge rendering if you'd want, but that's also a bit of new tech.

One solid change might be the `next/link` behavior change, but that even has an opt-out flag.

What's more interesting is how the concepts from pages dir to app dir have changed concretely, but have stayed the same in spirit. 

Anyhow, champion on with Next 13 and pages dir.
